sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/JREKeystoreTest.java

```diff
@@ -0,0 +1,280 @@
+        KeyFactory kf = KeyFactory.getInstance("RSA");
+        PKCS8EncodedKeySpec keySpecPKCS8 = new PKCS8EncodedKeySpec(Base64.getDecoder().decode(privateKey));
+        PrivateKey privKey = kf.generatePrivate(keySpecPKCS8);
+        return privKey;
+    }
+
+    private static Certificate[] getCertChain() {
+        try {
+            byte[] certificateBytes = Base64.getDecoder().decode(publicCert);
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
+            Certificate certificate = cf.generateCertificate(new ByteArrayInputStream(certificateBytes));
+            return new Certificate[]{certificate};
+        } catch (CertificateException e) {
+            throw new ProviderException(e);
+        }
+    }
+
+    private File getKeyStoreFile() {
+        String storePropName = PrivilegedActionImpl.privilegedGetProperty(
+            "javax.net.ssl.keyStore", jsseDefaultStore);
+
+        String[] fileNames =
+            new String[]{storePropName, defaultStore};
+        for (String fileName : fileNames) {
+            File f = new File(fileName);
+            if (f.isFile() && f.canRead()) {
+                return f;
+            }
+        }
+        return null;
+    }
+
+    private File getTrustStoreFile() {
+        String storePropName = PrivilegedActionImpl.privilegedGetProperty(
+            "javax.net.ssl.trustStore", jsseDefaultStore);
+        String[] fileNames =
+            new String[]{storePropName, defaultStore};
+        for (String fileName : fileNames) {
+            File f = new File(fileName);
+            if (f.isFile() && f.canRead()) {
+                return f;
+            }
+
+
+        }
+        return null;
+    }
+
+    private static class PrivateKeyStrategyImpl implements PrivateKeyStrategy {
+        @Override
+        public String chooseAlias(Map<String, PrivateKeyDetails> map, Socket socket) {
+            return "a-humble-key-entry-alias";
+        }
+
+    }
+
+
+    private static class PrivilegedActionImpl implements PrivilegedAction<String> {
+
+        private static String privilegedGetProperty(String theProp, String defaultVal) {
+            if (System.getSecurityManager() == null) {
+                String value = System.getProperty(theProp);
+                return (value == null || value.isEmpty()) ? defaultVal : value;
+            } else {
+                return AccessController.doPrivileged(
+                    new PrivilegedActionImpl(theProp, defaultVal));
+            }
+        }
+
+        private String theProp;
+        private String defaultVal;
+
+        private PrivilegedActionImpl(String theProp) {
+            this.theProp = theProp;
+
+        }
+
+        private PrivilegedActionImpl(String theProp, String defaultVal) {
+            this.theProp = theProp;
+            this.defaultVal = defaultVal;
+
+        }
+
+        public String run() {
+            String value = System.getProperty(theProp);
```

Comment:
`System.getProperty` need to be `doPrivileged`?

Reply:
Looks like yes if a security manager is installed.